databroker/broker.py

```diff
@@ -230,12 +232,28 @@ def clear_filters(self, **kwargs):
         self.filters.clear()
 
     def __getitem__(self, key):
-        """DWIM slicing
-
-        Some more docs go here
-        """
+        """Do-What-I-Mean slicing"""
         return search(key, self.mds)
 
+    def __getattr__(self, key):
+        try:
+            query = self.aliases[key]
+        except KeyError:
+            raise AttributeError(key)
+        if callable(query):
+            query = query()
+        return self(**query)
+
+    def alias(self, key, **query):
+        if hasattr(self, key) and key not in self.aliases:
+            raise ValueError("'%s' is not a legal alias." % key)
+        self.aliases[key] = query
+
+    def dynamic_alias(self, key, func):
+        if hasattr(self, key) and key not in self.aliases:
+            raise ValueError("'%s' is not a legal alias." % key)
+        self.aliases[key] = func
```

Comment:
Heh, duh. Yes. I was thinking, historydict might be overkill here anyway. People can set these up in their IPython profiles. I think the dynamic-ness is more useful than the automatic persistence. Especially since the "history" aspect of historydict is unlikely to be of much use here.